src/components/UserBtnsAndGrid/UserBtnsAndGrid.tsx

```diff
@@ -0,0 +1,46 @@
+'use client';
+
+import { useState } from 'react';
+import { BiShuffle } from 'react-icons/bi';
+import { LetterGrid } from '../LetterGrid/LetterGrid';
+import styles from './userBtnsAndGrid.module.scss';
+
+type UserBtnsAndGridProps = {
+  letters: string[];
+};
+
+export const UserBtnsAndGrid = ({ letters }: UserBtnsAndGridProps) => {
+  const [gameLetters, setGameLetters] = useState(letters);
+  const [wordInput, setWordInput] = useState('');
+
+  const resetWordInput = () => setWordInput(wordInput.slice(0, -1));
+
+  /* Durstenfeld shuffle algorithm (optimized version of Fisher-Yates) */
+  const handleShuffle = (arr: string[]) => {
```

Comment:
Maybe outsource this to the utilites folder and make a new file handleShuffle.ts
Handleshuffle should return arr.slice(0,1).concat(shuffleArr), so we can just pass that value into setGameLetters.

```js
const shuffleLetters = () => {
  const shuffled = handleShuffle(letters);
  setGameLetters(shuffled)
}

<button onClick={shuffleLetters} />
```